### PR TITLE
feat: inital project setup

### PR DESCRIPTION
**Pull request by aayushmau5**
**Description**

- Initalized NodeJS + Typescript Project(`package.json`, `package-lock.json` & `tsconfig.json`).
- Added eslint files.
- Added `jest` & `ts-jest` for testing.
- Added `semantic-release` package.

### CODE REVIEW

File: package.json
Comment (magicmatatjahu) [inline package.json:26]:
```suggestion
  "license": "Apache-2.0",
```

**Review by magicmatatjahu — COMMENTED** [review]
Only few comments :) Great job!

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
@aayushmau5 I am guessing you want this merged asap, and not wait for Lukasz, so I am just gonna help you move forward. Looking good 👍 

I would suggest adding a single `index.ts` in the `src` folder as the "initial version", that way package.json `main` file can be located 🙂 

We also need to trigger a manual release before merging this PR to make sure our workflow picks up the version.

@aayushmau5 do you have access to do this?

File: .eslintignore
Comment (magicmatatjahu) [reply .eslintignore:4]: Also I think that you should add in this file all content/list from `.gitignore` :)

File: .eslintrc
Comment (aayushmau5) [reply .eslintrc:12]: I don't think so. I check the typescript eslint plugin and they have `@typescript-eslint/recommended` plugin in their README.  

The reason I added those because both of those plugins were also present in `generator-model-sdk` repo.

I will remove the `eslint-recommended` plugin.

File: package.json
Comment (aayushmau5) [reply package.json:10]: Oops. I overlooked this. Will fix. :D

**Comment by aayushmau5** [comment]
> We also need to trigger a manual release before merging this PR to make sure our workflow picks up the version.
> @aayushmau5 do you have access to do this?

I don't think so. (I'm not sure)

**Comment by jonaslagoni** [comment]
> > We also need to trigger a manual release before merging this PR to make sure our workflow picks up the version.
> > @aayushmau5 do you have access to do this?
> 
> I don't think so. (I'm not sure)

Do you see a `Create a new release` under the main repo (https://github.com/asyncapi/diff) on the right side under `Releases`?

**Comment by aayushmau5** [comment]
Nope.

**Comment by jonaslagoni** [comment]
> Nope.

Alright, we need @derberg to add you as code owner once he gets back. Feel free to ping him with the request already so he is aware of it when he gets back 🙂   

I will create the release so we can merge the PR once the change requests are addressed 👍

**Comment by aayushmau5** [comment]
> Alright, we need @derberg to add you as code owner once he gets back. Feel free to ping him with the request already so he is aware of it when he gets back.

Will do :smile: 

> I will create the release so we can merge the PR

Thanks :pray:

**Comment by magicmatatjahu** [comment]
@aayushmau5 You probably need to wait for review from your Postman mentor :) Ping us when you want to merge it :)

**Comment by jonaslagoni** [comment]
@aayushmau5 can you change the PR title to use [conventional commit syntax](https://www.conventionalcommits.org/en/v1.0.0/#summary)? It has to be similar to `feat: initial project setup` That way the automated workflow will bump the version in package.json and add a new release on NPM so it goes from 0.0.1 -> 0.1.0

**Review by vinitshahdeo — APPROVED** [review]
@aayushmau5 LGTM. 👏 
@jonaslagoni we can merge it now.

**Review by jonaslagoni — APPROVED** [review]

**Pull request merged**
